tests/data_loading/test_data_manager.py

```diff
@@ -33,10 +34,12 @@ def test_base_datamanager(self):
 
         batch_dict = {"data": np.asarray(data), "label": np.asarray(labels)}
 
-        for key, val in next(manager.get_batchgen()).items():
+        augmenter = manager.get_batchgen()
+        augmenter_iter = iter(augmenter)
+        for key, val in next(augmenter_iter).items():
             self.assertTrue((val == batch_dict[key]).all())
 
-        for key, val in next(manager.get_batchgen()).items():
+        for key, val in next(augmenter_iter).items():
             self.assertEqual(len(val), batch_size)
 
 
```

Comment:
I would like to propose to add some additional integration tests here.

2 Tests: Check with mock object if transforms are called correctly. (serial and parallel)

Additional Tests: Create a dataset with an internal ordering (something like an ID) and check if the sampling order is correct. This should be done with a sequential sampler and a random sampler (serial and parallel). The random sampler should be tested with a seed and check the first few sampler + sampling of all indices (drop_last should be false).

Reply:
I added tests for sampling order, but IMO the tests with random sampler and indices do not belong here, but to the sampler tests (if anywhere). But since we use well tested libraries for sampling, I don't think we need them at all.

Reply:
I thought one of the main problems solved in this PR is the correct sampling with the parallel augmenter (the whole purpose of the pipe construct). Therefore, there should be a test where the correctness of this is assessed.

Reply:
We had correct sampling before. The main problem solved here is the speed. But I agree that in the long run, we should have a test for this. Are you ok with postponing this in a separate Issue too or do you want to have it included here?

Reply:
We can add it to a separate Issue, just wanted to mention it for the future :)